### PR TITLE
Fix some nits in the documentation

### PR DESCRIPTION
**Pull request by mhuin**
Just a few things to clarify.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -7,11 +7,16 @@ understand the choices made by the project.
 
 ## Run the APIv1 test (Python)
 
+:warning: Make sure to follow first the installation instructions in the README!
```
Comment (TristanCacqueray) [inline CONTRIBUTING.md:10]: What is missing if you don't follow the instructions? I wonder if we should bother with installing the service for running those tests.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -7,11 +7,16 @@ understand the choices made by the project.
...
+
 Tests rely on the Elasticsearch service so first you need to ensure the Elasticsearch container is running.
-You can run the "docker-compose up -d" or you can only run the Elasticsearch container by running "docker-compose start elastic".
+You can run the "docker-compose -f docker-compose.yml.dev up -d" or you can only run the Elasticsearch container by running "docker-compose -f docker-compose.yml.dev start elastic".
+
+(Or see below to use a symlink to the development compose instead instead)
```
Comment (TristanCacqueray) [inline CONTRIBUTING.md:15]: Similarly, I think the test environment should be isolated from the install. Perhaps we should drop the docker-compose instruction and keep the podman (or docker) command below.

**Review by TristanCacqueray — CHANGES_REQUESTED** [review]

**Comment by morucci** [comment]
I'll takeover this change to refine it and remove the docker-compose.dev.yml support.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -177,36 +134,42 @@ cabal repl monocle
 λ> withClient "http://127.0.0.1:8081" Nothing $ \client -> runMacroscope True "../etc/config.yaml" 30 client
 ```
 
-### Running the services manually using nix
+### Running the services manually using NIX
 
 This section describes how to start the Monocle services directly on your host using nix.
 
-#### Elastic
+Note that the above commands can be started in emacs buffer using:
```
Comment (TristanCacqueray) [inline CONTRIBUTING.md:141]:
```suggestion
Note that the commands below can be started in emacs buffer using:
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -230,15 +193,40 @@ nix-shell --command monocle-api2-start
 λ> withClient "http://localhost:18080" Nothing $ \client -> runMacroscope True "../etc/config.yaml" 30 client
 ```
 
-#### Launch with emacs lisp
+## Contributing a new driver
+
+We do not provide specific documentation to cover that topic yet but the source code of
+the [GitLab driver](monocle/haskell/src/Lentille/GitLab/MergeRequests.hs) might be a good
+source of knowledge to hack on a new crawler.
```
Comment (TristanCacqueray) [inline CONTRIBUTING.md:200]:
```suggestion
There is no specific documentation to cover that topic yet but the source code of
the [GitLab driver](monocle/haskell/src/Lentille/GitLab/MergeRequests.hs) might be a good
source of knowledge to hack on a new crawler.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,99 +1,42 @@
...
-```ShellSession
-$ tox
-```
+This document provides some instructions to get started with Monocle development. It covers topics
+such as running tests, running services w/o docker-compose or running the codegen.
 
-## Update the APIv2 (protobuf)
+As explain in the [README](README.md), Monocle is composed of several services and during
+our current refactoring some services run though the Python runtime (legacy components) and other
+are built based on Haskell.
```
Comment (TristanCacqueray) [inline CONTRIBUTING.md:8]:
```suggestion
As explained in the [README](README.md), Monocle is composed of several services, some services run though the Python runtime (legacy components) and the new one are built using the Haskell runtime.
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -131,25 +74,26 @@ server {
 }
 ```
 
-#### Elastic
+Then ensure NGINX service is started/reloaded on your system.
+
+#### ElasticSearch
+
+Monocle relies on ElasticSearch as backend to store Changes and Change' events.
```
Comment (TristanCacqueray) [inline CONTRIBUTING.md:81]:
```suggestion
Monocle relies on ElasticSearch as backend to store Changes events.
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -131,25 +74,26 @@ server {
...
 ./contrib/start-elk.sh 9200
 ```
 
-#### APIv1
+#### API
 
-```ShellSession
-./contrib/start-apiv1.sh 9878
-```
+To start the Monocle API then run the following commands.
```
Comment (TristanCacqueray) [inline CONTRIBUTING.md:89]:
```suggestion
To start the Monocle API run the following commands.
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -131,25 +74,26 @@ server {
...
 
-#### Web
+To reload modules when code is updated then type `:reload`.
 
-```ShellSession
-./contrib/start-web.sh
-```
+In addition you could run `ghcid -c 'cabal repl monocle'` in another terminal, then on every save in your IDE you'll see a list of the errors and warnings from ghc.
```
Comment (TristanCacqueray) [inline CONTRIBUTING.md:93]: Perhaps we don't need to bother with `ghcid`, I think the ghci usage is good enough, then interested users are probably aware of ghcid, or other editor integrations.

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -131,25 +74,26 @@ server {
...
 
-#### APIv2
+You might need to restart the repl and/or ghcid when new dependencies are
+added in the `monocle.cabal` file.
```
Comment (TristanCacqueray) [inline CONTRIBUTING.md:96]:
```suggestion
> The REPL needs to be restarted when the `monocle.cabal` file is modified.
```

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -159,14 +103,27 @@ cabal repl monocle
 λ> run 9879 "http://localhost:9200" "../etc/config.yaml"
 ```
 
+#### Web
+
+The Monocle React WebAPP (hot reload is enabled).
```
Comment (TristanCacqueray) [inline CONTRIBUTING.md:108]:
```suggestion
Run this command to start the Monocle React live server with hot reload.
```

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -230,15 +193,40 @@ nix-shell --command monocle-api2-start
...
+
+## Running tests
 
-The above commands can be started in emacs buffer using:
+Tests rely on the Elasticsearch service so first you need to ensure the ElasticSearch is running on your system. To start the service use the script `contrib/start-elk.sh` or the
+related nix-shell command.
+
+### On Python code base (legacy)
+
+Tests can be executed using (ajust the port the 19200 if you used the nix-shell command to start Elastic):
```
Comment (TristanCacqueray) [inline CONTRIBUTING.md:209]:
```suggestion
Tests can be executed using:
```

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -230,15 +193,40 @@ nix-shell --command monocle-api2-start
...
 
 ```ShellSession
-nix-shell --command launch-monocle-with-emacs
+ELASTIC_CONN=127.0.0.1:9200 tox
 ```
 
-## Contributing a new driver
+### On the Haskell code base
+
+Tests can be executed using (ajust the port the 19200 if you used the nix-shell command to start Elastic):
```
Comment (TristanCacqueray) [inline CONTRIBUTING.md:217]:
```suggestion
Tests can be executed using:
```

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -177,36 +134,42 @@ cabal repl monocle
...
 ```ShellSession
 nix-shell --command nginx-start
 ```
 
-#### APIv1
+#### ElasticSearch
 
 ```ShellSession
-nix-shell --command monocle-api-start
+nix-shell --command elk-start
```
Comment (TristanCacqueray) [inline CONTRIBUTING.md:156]: Let's add here: `export ELASTIC_CONN=127.0.0.1:19200`

File: CONTRIBUTING.md [hunk 14]
```diff
@@ -230,15 +193,40 @@ nix-shell --command monocle-api2-start
...
+Tests rely on the Elasticsearch service so first you need to ensure the ElasticSearch is running on your system. To start the service use the script `contrib/start-elk.sh` or the
+related nix-shell command.
+
+### On Python code base (legacy)
+
+Tests can be executed using (ajust the port the 19200 if you used the nix-shell command to start Elastic):
 
 ```ShellSession
-nix-shell --command launch-monocle-with-emacs
+ELASTIC_CONN=127.0.0.1:9200 tox
```
Comment (TristanCacqueray) [inline CONTRIBUTING.md:212]:
```suggestion
tox
```

File: CONTRIBUTING.md [hunk 15]
```diff
@@ -230,15 +193,40 @@ nix-shell --command monocle-api2-start
...
-## Contributing a new driver
+### On the Haskell code base
+
+Tests can be executed using (ajust the port the 19200 if you used the nix-shell command to start Elastic):
 
-Please refer to the file `dummy/change.py` that contain a Dummy driver and
-some comments to help you get started.
+```ShellSession
+cd haskell
+ELASTIC_URL=http://localhost:9200 cabal test
```
Comment (TristanCacqueray) [inline CONTRIBUTING.md:221]:
```suggestion
cabal test
```

File: CONTRIBUTING.md (hunk 8 above)
Comment (morucci) [reply CONTRIBUTING.md:93]: I'll prefer to keep it if you don't mind to much.

**Review by TristanCacqueray — APPROVED** [review]

**Pull request merged**